### ISSUE TITLE
Theory Question: Numerical Advantage of Updating Overlapping Cells Together

### ISSUE THREAD

**Issue by MaximilianHoffmann** [denisecailab/minian#181]
In the docs you write:

> Although we obtained YrA for each unit, it is still beneficial to group together units that heavily overlap, and carry out the update process group-wise instead of independently for each unit. This way the relative numerical relationship between heavily-overlapping units are better preserved. For this purpose, we compute pairwise Jaccard index for all units, which serve as a metric of overlap between spatial footprints of units. The parameter jac_thres is the threshold of Jaccard index above which units will be grouped together transitively.

and indeed this is, what Caiman is doing as well. I don't quite understand the reason for this. As I understand it Yra is never updated within one temporal iteration, right?  So how do these different optimization problems depend on each other?
Thanks for any help and sorry if this issue tracker is not quite the right place for this.

**Comment by phildong** [comment]
You are correct that `YrA` is not updated at all, it serve as the input/data the model is fitting to.
When updating multiple cells together, the input change from 1d (time) to 2d (cell x time) and a single optimization is performed with this 2d matrix of multiple cells.
In theory this can lead to more accurate estimation of spikes for overlapping cells, especially when some cells are much brighter than others.
